src/Simulation/Core/IQuantumMachine.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Runtime
+{
+    /// <summary>
+    /// Interface that a quantum machine must implement.
+    /// </summary>
+    /// <typeparam name="Job">Type of job that handles the execution of a program in the quantum machine.</typeparam>
+    /// <typeparam name="QRawResult">Type of result the quantum machine returns.</typeparam>
+    public interface IQuantumMachine<Job, QRawResult>
```

Comment:
It might be worth it to conform to the type parameters name convention in C# and change the type parameter names to start with T instead of Q.